src/lists.py

```diff
@@ -42,17 +42,27 @@ def print_list(x):
 match sys.argv[1]:
     case "mean":
         # put your solution to the first exercise here
-        mean = "mean of x"
+        sum = 0 
+        for i in x: 
+            sum=sum+i
+        mean = sum/len(x)
         print(mean)
 
     case "times":
         # Put your solution to the second exercise here
         times_three = []
+        for i in x:
+            times_three.append(i*3)
```

Comment:
Perfect.

When things are going well, we have to increase the difficulty. There is a thing called "list comprehension" in Python. If you write

```python
[ f(a) for a in x if g(a) ]
```

you get a list of `f(a)` (some expression involving `a`) for each `a` in `x` where `g(a)` (some other expression) is true. You can leave out the `if g(a)` part.

With that, for example

```python
times_three = [i*3 for i in x]
```

solves the same problem you just did. It is in every way the same loop; it creates a list and appends to it, so it doesn't add any functionality to loops, but it is more consise.

Try to write some code that use list comprehension.